### ISSUE TITLE
Feature request: "Missing Cover Art" Smart Collection

### ISSUE THREAD

**Issue by cabel** [OpenEmu/OpenEmu#2270]
**Request:** I'd love to have a Smart Collection of games that are just games missing cover art. This would make it far easier for me to complete my art collection, something that I think I'm physically incapable of not attempting.

**Bonus Request:** Hold option while right clicking on a game and "Download Cover Art" becomes "Google Image Search", kicking off Safari with a Google Image Search of the selected game's name???

Thank you for your amazing work on this app, easily one of my favorite on the Mac!

**Comment by clobber** [comment]
First one kinda falls under both #664 and #811, so tagging it here.

Second one, naaah, want to keep UI clutter down. But interesting idea.

Thanks!

**Comment by cabel** [comment]
But… but… if it's just an option-key menu swap there's no UI clutter! :)

But thank you for considering it! And thanks for your hard work!

**Comment by clobber** [comment]
Okay :P We'll take a look but let me know if you have any other cover art lookup issues so I can clear some things up.

Our internal database only tracks retail "good" dumps of the games, so no translations, overdumps, hacks, betas -- that sort of thing.

I guess my gripe with the option-click is it's not really discoverable either. Maybe an actual context menu for ones with the missing covers would be a decent compromise.

**Comment by diotro** [comment]
Even the reverse of this "Smart Collection" where instead the games that DO have artwork are listed. Once you have a big collection going, the standard color bars begin to become a little bit of an eyesore.

**Comment by cabel** [comment]
I know what you mean about Option-Click, although Apple does use it extensively. (Option-Right-Click on a file to swap "Copy [File]" to "Copy [File] as Pathname"! Very useful!)

I think you're right, a context menu on a missing cover art is the perfect compromise.

Also, I am considering filing a separate bug on this, tell me if it's worth it:

It would be _really_ nice to know if a blank cover art is because of a bad dump, or missing art in the database. Is there any way to determine this today?

**Comment by clobber** [comment]
@cabel Eh not worthy of a bug request. So, missing cover art simply means it did not return a match in the database for various possible reasons:

1. Your IP is blocked from the source.
2. Cover art did not exist from the source (gamefaqs).
3. The game matched but the particular game didn't have a URL to cover art added yet (incomplete).
4. The system for those games has no cover art added yet (incomplete).
5. The game was not matched in the database at all.

This has reminded me to update our Cover Art guide so I will elaborate on this there: https://github.com/OpenEmu/OpenEmu/wiki/User-guide:-Box-art

If you have a specific case that you think should be returning art but doesn't, let me know and I'll take a look.

**Comment by cabel** [comment]
Hey Clobber,

Thanks for the info and help as always!

Here's my take. **Overloading 5 different possible failure states into a single UI element — the "color bars" default artwork — means I can't make the right decision on how to "fix" the problem for that game.**

Do I have a bad dump? I'll want to download a better one. Is the art missing from the source? I'll find art. But I don't know what action to take, because I don't know what went wrong. And man, if my IP is blocked, I _really_ want to know that!

Even something as simple as a tooltip on hover over the missing art — "Cover art did not exist" or "Game not matched" — would go a LONG way to helping me clean up my collection.

I hope this makes sense!

Three other notes, that I can file into other bugs if you want:

• Today on a whim I did "Download Cover Art" on a  NES game that previously didn't get cover art, and suddenly the cover art popped up! This worked for a bunch more. Now I'm paranoid — how often should I attempt this on games that couldn't download art before?

• Knowing that more art might exist now, I did a "Select All" on my NES games and did "Download Cover Art". I was surprised it started to re-download covers that I already had. This goes back to my original bug — if I could highlight only titles with missing art, I wouldn't have wasted so much of your bandwidth re-downloading art I had.

• Have you considered something like art submission? I would LOVE to help your art collection grow. There's overhead and maintenance involved, but this is why CDDB worked and grew so huge — user contributions.
